### PR TITLE
RRule: Include set timezone in serialized result

### LINKED ISSUE #37: `RRule` does not include set timezone during serialization

### PR DESCRIPTION
**Pull request by nilmerg**
fixes #37
blocked by #39

### CODE REVIEW

**Comment by nilmerg** [comment]
Partly obsolete since #40, but for `UNTIL` still relevant.

File: tests/RRuleTest.php
Comment (yhabteab) [inline tests/RRuleTest.php:1]: [As far as I understand](http://php.adamharvey.name/manual/en/function.date-default-timezone-set.php), `date_default_timezone_set();` only sets the timezone for the current script, so I don't think that you need the `try...finally` to reset the timezone.

File: tests/RRuleTest.php [hunk 1]
```diff
@@ -156,4 +156,33 @@ public function testJsonSerializeAndDeserialize()
...
+        date_default_timezone_set('Europe/Berlin');
+
+        try {
+            $startWithTz = new DateTime('01-06-2023T12:00:00', new DateTimeZone('America/New_York'));
+            $endWithTz = new DateTime('01-06-2024T07:00:00', new DateTimeZone('America/New_York'));
+
+            $rrule = RRule::fromJson(json_encode(RRule::fromFrequency(RRule::DAILY)
+                ->startAt($startWithTz)
+                ->endAt($endWithTz)
+                ->jsonSerialize()));
```
Comment (yhabteab) [inline tests/RRuleTest.php:172]:
```suggestion
            $rrule = RRule::fromJson(
                json_encode(
                    RRule::fromFrequency(RRule::DAILY)
                        ->startAt($startWithTz)
                        ->endAt($endWithTz)
                        ->jsonSerialize()
                )
            );
```

**Review by yhabteab — CHANGES_REQUESTED** [review]

File: tests/RRuleTest.php
Comment (nilmerg) [reply tests/RRuleTest.php:1]: *the current script* is the entire test run. i.e. all tests combined.

**Review by yhabteab — APPROVED** [review]

**Pull request merged**